### PR TITLE
Added example for the new async parameters API

### PR DESCRIPTION
**Pull request by esteve**
Connects to ros2/ros2#28

@dirk-thomas @tfoote @wjwwood

### CODE REVIEW

File: userland/src/parameters.cpp [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  auto f1 = parameters_client->set_parameters(
+    parameters, [](std::shared_future<std::vector<rcl_interfaces::ParameterSetResult>> f) {
+    for(auto v : f.get()) {
+      if(!v.successful) {
+        std::cerr << v.reason << std::endl;
+      }
+    }
+  });
+
+  rclcpp::spin_until_future_complete<std::vector<rcl_interfaces::ParameterSetResult>>(node, f1);
```
Comment (tfoote) [inline userland/src/parameters.cpp]: Is there any way we can get this to automatically infer the template type?

**Comment by tfoote** [comment]
One question but lgtm

File: userland/src/parameters.cpp (hunk 1 above)
Comment (esteve) [inline userland/src/parameters.cpp]: I've changed the example to use the inferred template parameter.

Comment (tfoote) [inline userland/src/parameters.cpp]: Nice! That's very clean now.

**Comment by dirk-thomas** [comment]
The async API of `AsyncParametersClient` lgtm.

Should the example also contain some API calls to use the results of the futures to see how a user accesses the data? WIth all the `auto` *magic* it is less clear what to do with `fN` from reading the example.

Nitpicking:
* copyright year
* unused includes
* unnecessary `std::string` for `ParameterVariant` constructor parameters

**Pull request merged**